Firestore/src/V1/FirestoreClient.php

```diff
@@ -26,13 +26,139 @@
+     * a database_root resource.
+     *
+     * @param string $project
+     * @param string $database
+     *
+     * @return string The formatted database_root resource.
+     * @deprecated
+     */
+    public static function databaseRootName($project, $database)
+    {
+        return (new PathTemplate('projects/{project}/databases/{database}'))->render([
+            'project' => $project,
+            'database' => $database,
+        ]);
+    }
+
+    /**
+     * Formats a string containing the fully-qualified path to represent
+     * a document_path resource.
+     *
+     * @param string $project
+     * @param string $database
+     * @param string $documentPath
+     *
+     * @return string The formatted document_path resource.
+     * @deprecated
+     */
+    public static function documentPathName($project, $database, $documentPath)
+    {
+        return (new PathTemplate('projects/{project}/databases/{database}/documents/{document_path=**}'))->render([
+            'project' => $project,
+            'database' => $database,
+            'document_path' => $documentPath,
+        ]);
+    }
+
+    /**
+     * Formats a string containing the fully-qualified path to represent
+     * a document_root resource.
+     *
+     * @param string $project
+     * @param string $database
+     *
+     * @return string The formatted document_root resource.
+     * @deprecated
+     */
+    public static function documentRootName($project, $database)
+    {
+        return (new PathTemplate('projects/{project}/databases/{database}/documents'))->render([
+            'project' => $project,
+            'database' => $database,
+        ]);
+    }
+
+    /**
+     * Parses a formatted name string and returns an associative array of the components in the name.
+     * The following name formats are supported:
+     * Template: Pattern
+     * - anyPath: projects/{project}/databases/{database}/documents/{document}/{any_path=**}
+     * - databaseRoot: projects/{project}/databases/{database}
+     * - documentPath: projects/{project}/databases/{database}/documents/{document_path=**}
+     * - documentRoot: projects/{project}/databases/{database}/documents.
+     *
+     * The optional $template argument can be supplied to specify a particular pattern, and must
+     * match one of the templates listed above. If no $template argument is provided, or if the
+     * $template argument does not match one of the templates listed, then parseName will check
+     * each of the supported templates, and return the first match.
+     *
+     * @param string $formattedName The formatted name string
+     * @param string $template      Optional name of template to match
+     *
+     * @return array An associative array from name component IDs to component values.
+     *
+     * @throws ValidationException If $formattedName could not be matched.
+     * @deprecated
+     */
+    public static function parseName($formattedName, $template = null)
```

Comment:
It's strange to me that a method with this much logic in it was being generated as part of GAPIC. Any idea how this was being added previously and why it was removed @dwsupplee ?

Reply:
I'm surprised to see it looks like all the path helpers on the firestore client are missing, my guess as to why this specific function was removed is because there may be logic in the generator that only generates this method given there are any path helpers generated as well.

I'm taking a look to better understand what happened to all the helpers.

Reply:
I've only seen a method like this in the Firestore client, fortunately, so I think this one-off fix is sufficient.